### DISCUSSION TITLE
HomeAssistant Integration

### DISCUSSION THREAD

**Discussion by icemansid** [David00/rpi-power-monitor#70] [Ideas]
I currently have this system and have been using it since its inception. Love it.  Is there any chance there is a way it can be integrated into HomeAssistant to get power readings there? Not just influx data but add it as a HomeAssistant sensor.

**Comment by David00** [comment]
Hey @icemansid, awesome!  

So I'm not too familiar with Home Assistant (never personally deployed one myself), but I have gotten this question a few times. There's a script made by someone else to get data from this project into HA. I can't personally vouch for it, but check it out and let me know what you think!
https://github.com/heytensai/rpi-power-monitor-mqtt

I'm open to exploring HA in more detail though and possibly bringing native support for HA into the project.

**Reply by icemansid** [reply]
Man that was super fast! Thanks for the quick reply and script. MQTT is really the "answer" as MQTT is pretty common in Home Assistant. Highly recommend trying it out.

**Comment by David00** [comment]
There have been a couple more new community contributions that can get you up and running with Home Assistant. Here is an updated list containing all the options (including the one mentioned above).

* [General MQTT export script](https://github.com/heytensai/rpi-power-monitor-mqtt), by @heytensai
* [Full guide](https://github.com/emersonsc/rpi-power-monitor/tree/master/Home%20Assistant) by @emersonsc
* [MQTT plugin for this project](https://github.com/David00/rpi-power-monitor/pull/104/commits/82a0a7b408edc0dbc82446d25290e1b8375a6817), by @jmadden91  (this plugin will be listed on the [plugins page in the documentation](https://david00.github.io/rpi-power-monitor/docs/v0.3.0/plugin-list.html), once I can get a completed README template from the author)
